<launch>
  <!-- Arguments -->
  <arg name="cmd_vel_topic" default="/cmd_vel2" />
  <arg name="odom_topic" default="odom" />

  <!-- move_base -->
  <node pkg="move_base" type="move_base" respawn="false" name="move_base" output="screen">
    <param name="base_local_planner" value="teb_local_planner/TebLocalPlannerROS"/>
    <param name="base_global_planner" value="global_planner/GlobalPlanner" />
    
    <rosparam file="$(find gazebo_nav)/launch/config/costmap_common_params.yaml" command="load" ns="global_costmap" />
    <rosparam file="$(find gazebo_nav)/launch/config/costmap_common_params.yaml" command="load" ns="local_costmap" />
    <rosparam file="$(find gazebo_nav)/launch/config/move_base_params.yaml" command="load" />
    
    <rosparam file="$(find gazebo_nav)/launch/config/local_costmap_params.yaml" command="load" />
    <rosparam file="$(find gazebo_nav)/launch/config/global_costmap_params.yaml" command="load" />

    <rosparam file="$(find gazebo_nav)/launch/config/global_planner_params.yaml" command="load" />
    <rosparam file="$(find gazebo_nav)/launch/config/teb_local_planner_params.yaml" command="load" />

    <remap from="cmd_vel" to="$(arg cmd_vel_topic)"/>
    <remap from="odom" to="$(arg odom_topic)"/>
  </node>
</launch>
